launch_frontend/launch_frontend/action_parse_methods.py

```diff
@@ -0,0 +1,117 @@
+    if cwd is not None:
+        kwargs['cwd'] = parser.parse_substitution(cwd)
+    name = entity.get_attr('name', optional=True)
+    if name is not None:
+        kwargs['name'] = parser.parse_substitution(name)
+    prefix = entity.get_attr('launch-prefix', optional=True)
+    if prefix is not None:
+        kwargs['prefix'] = parser.parse_substitution(prefix)
+    output = entity.get_attr('output', optional=True)
+    if output is not None:
+        kwargs['output'] = output
+    shell = entity.get_attr('shell', types='bool', optional=True)
+    if shell is not None:
+        kwargs['shell'] = shell
+    # Conditions won't be allowed in the `env` tag.
+    # If that feature is needed, `set_enviroment_variable` and
+    # `unset_enviroment_variable` actions should be used.
+    env = entity.get_attr('env', types='list[Entity]', optional=True)
+    if env is not None:
+        # TODO(ivanpauno): Change `ExecuteProcess` api. `additional_env`
+        # argument is supposed to be a dictionary with `SomeSubstitutionType`
+        # keys, but `SomeSubstitutionType` is not always hashable.
+        # Proposed `additional_env` type:
+        #   Iterable[Tuple[SomeSubstitutionType, SomeSubstitutionsType]]
+        env = {e.get_attr('name'): parser.parse_substitution(e.get_attr('value')) for e in env}
+        kwargs['additional_env'] = env
+    args = entity.get_attr('args', optional=True)
+    # `args` is supposed to be a list separated with ' '.
+    # All the found `TextSubstitution` items are split and
+    # added to the list again as a `TextSubstitution`.
+    # TODO(ivanpauno): Change `ExecuteProcess` api from accepting
+    # `Iterable[SomeSubstitutionType]` `cmd` to `SomeSubstitutionType`.
+    # After performing the substitution in `cmd`, shlex.split should be done.
+    # This will also allow having a substitution which ends in more than one
+    # argument.
+    if args is not None:
+        args = parser.parse_substitution(args)
+        new_args = []
+        for arg in args:
+            if isinstance(arg, TextSubstitution):
+                text = arg.text
+                text = shlex.split(text)
+                text = [TextSubstitution(text=item) for item in text]
+                new_args.extend(text)
+            else:
+                new_args.append(arg)
+        args = new_args
+    else:
+        args = []
+    cmd_list = [cmd]
+    cmd_list.extend(args)
+    if_cond = entity.get_attr('if', optional=True)
+    unless_cond = entity.get_attr('unless', optional=True)
+    if if_cond is not None and unless_cond is not None:
+        raise RuntimeError("if and unless conditions can't be usede simultaneously")
+    if if_cond is not None:
+        kwargs['condition'] = IfCondition(predicate_expression=if_cond)
+    if unless_cond is not None:
+        kwargs['condition'] = UnlessCondition(predicate_expression=unless_cond)
+
+    return launch.actions.ExecuteProcess(
+        cmd=cmd_list,
+        **kwargs
+    )
```

Comment:
If we finally modify the `expose_*` decorators to automatically infer the parsing function from the constructor, this will automatically be solved (i.e.: the decorator should not only check the type annotation of the constructor arguments, but it should also check the derived class constructor arguments).
For the moment, I will let this as-is.

Reply:
Finally, I have corrected it!

Reply:
> Finally, I have corrected it!

Awesome! And yes, automatic inference may reduce the need for this feature somewhat but we don't plan to work on it soon and we don't know if it's feasible for every use case.